app/src/main/java/io/realm/android/SecureUserStore.java

```diff
@@ -22,151 +6,97 @@
  *
- * This throws a {@link KeyStoreException} in case of an error or KeyStore being unvailable (unlocked).
+ * This throws a {@link KeyStoreException} in case of an error or KeyStore being unavailable (unlocked).
  *
  * See also: io.realm.internal.android.crypto.class.CipherClient
  * @see <a href="https://developer.android.com/training/articles/keystore.html">Android KeyStore</a>
  */
-public class SecureUserStore implements UserStore {
-    private static final String REALM_OBJECT_SERVER_USERS = "realm_object_server_users";
+public class SecureUserStore extends ObjectStoreUserStore {
     private final CipherClient cipherClient;
-    private final SharedPreferences sp;
-    private SyncUser cachedCurrentUser; // Keep a quick reference to the current user
 
     public SecureUserStore(final Context context) throws KeyStoreException {
+        super(context.getFilesDir().getPath());
         cipherClient = new CipherClient(context);
-        sp = context.getSharedPreferences(REALM_OBJECT_SERVER_USERS, Context.MODE_PRIVATE);
     }
 
     /**
      * Store user as serialised and encrypted (Json), inside the private {@link SharedPreferences}.
-     * @param key the {@link SharedPreferences} key.
      * @param user we want to save.
      * @return The previous user saved with this key or {@code null} if no user was replaced.
      */
     @Override
-    public SyncUser put(String key, SyncUser user) {
-        String previousUser = sp.getString(key, null);
-        SharedPreferences.Editor editor = sp.edit();
-        String userSerialisedAndEncrypted;
+    public void put(SyncUser user) {
         try {
-            userSerialisedAndEncrypted = cipherClient.encrypt(user.toJson());
+            String userSerialisedAndEncrypted = cipherClient.encrypt(user.toJson());
+            updateOrCreateUser(user.getIdentity(), userSerialisedAndEncrypted, "");
         } catch (KeyStoreException e) {
-            e.printStackTrace();
-            return null;
-        }
-        editor.putString(key, userSerialisedAndEncrypted);
-        // Optimistically save. If the user isn't saved due to a process crash it isn't dangerous.
-        editor.apply();
-
-        if (UserStore.CURRENT_USER_KEY.equals(key)) {
-            cachedCurrentUser = user;
-        }
-        if (previousUser != null) {
-            try {
-                String userSerialisedAndDecrypted = cipherClient.decrypt(previousUser);
-                return SyncUser.fromJson(userSerialisedAndDecrypted);
-            } catch (KeyStoreException e) {
-                e.printStackTrace();
-                return null;
-            }
-        } else {
-            return null;
+            RealmLog.error(e);
         }
     }
 
     /**
      * Retrieves the {@link SyncUser} by decrypting first the serialised Json.
-     * @param key the {@link SharedPreferences} key.
      * @return the {@link SyncUser} with the given key.
```

Comment:
I think you reviewed the previous commit `ca72a833f86589b6e5c3765382c9d7def7e86bcd`  for this file 😀